Framework/Algorithms/test/CreateMonteCarloWorkspaceTest.h

```diff
@@ -0,0 +1,169 @@
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAlgorithms/CreateMonteCarloWorkspace.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidFrameworkTestHelpers/WorkspaceCreationHelper.h"
+#include "MantidHistogramData/Histogram.h"
+#include <cxxtest/TestSuite.h>
+
+#include <random>
+
+using namespace Mantid::Algorithms;
+using namespace Mantid::API;
+using namespace Mantid::HistogramData;
+
+using Mantid::Algorithms::CreateMonteCarloWorkspace;
+
+class CreateMonteCarloWorkspaceTest : public CxxTest::TestSuite {
+public:
+  static CreateMonteCarloWorkspaceTest *createSuite() { return new CreateMonteCarloWorkspaceTest(); }
+  static void destroySuite(CreateMonteCarloWorkspaceTest *suite) { delete suite; }
+
+  void test_Init() {
+    CreateMonteCarloWorkspace alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+    TS_ASSERT(alg.isInitialized());
+  }
+
+  void test_computeNumberOfIterations() {
+    CreateMonteCarloWorkspace alg;
+    Mantid::HistogramData::HistogramY yData = {1.0, 2.0, 3.0, 4.0};
+    int iterations = alg.computeNumberOfIterations(yData, 0);
+    TS_ASSERT_EQUALS(iterations, 10); // Verify yData rounds correctly
+
+    // Test custom Monte Carlo events
+    iterations = alg.computeNumberOfIterations(yData, 20);
+    TS_ASSERT_EQUALS(iterations, 20); // Should override with the custom number
+  }
+
+  void test_computeNormalizedCDF() {
+    CreateMonteCarloWorkspace alg;
+    Mantid::HistogramData::HistogramY yData = {1.0, 2.0, 3.0, 4.0};
+    std::vector<double> cdf = alg.computeNormalizedCDF(yData);
+    TS_ASSERT_EQUALS(cdf.size(), yData.size());
+    TS_ASSERT_DELTA(cdf.back(), 1.0, 1e-6); // Check the last element is normalized to 1.0
+  }
+
+  void test_fillHistogramWithRandomData() {
+    CreateMonteCarloWorkspace alg;
+    std::vector<double> cdf = {0.1, 0.3, 0.6, 1.0};
+    Mantid::API::Progress progress(nullptr, 0.0, 1.0, 1); // Dummy progress
+    Mantid::HistogramData::HistogramY outputY = alg.fillHistogramWithRandomData(cdf, 100, 32, progress);
+
+    auto sumCounts = std::accumulate(outputY.begin(), outputY.end(), 0.0);
+    TS_ASSERT_EQUALS(sumCounts, 100); // Ensure the total number of counts is correct
+  }
+
+  void test_scaleInputToMatchMCEvents() {
+    CreateMonteCarloWorkspace alg;
+    Mantid::HistogramData::HistogramY yData = {1.0, 2.0, 3.0, 4.0};
+
+    // Test scaling
+    int targetMCEvents = 20;
+    Mantid::HistogramData::HistogramY scaledY = alg.scaleInputToMatchMCEvents(yData, targetMCEvents);
+
+    double totalScaledCounts = std::accumulate(scaledY.begin(), scaledY.end(), 0.0);
+    TS_ASSERT_DELTA(totalScaledCounts, 20.0, 1e-6); // Verify the scaled sum matches targetMCEvents
+  }
+
+  MatrixWorkspace_sptr createInputWorkspace(int numBins, double initialValue) {
```

Comment:
Looks like there are a few helper functions here, these are good but perhaps separate them from the tests at the bottom of the file? Makes it easier to read somehow (perhaps just personal preference)